### PR TITLE
Pending invitations modal

### LINKED ISSUE #479: Pending invitations modal
### Summary and Background
Pending invitations modal allows Super Admin to view/delete account invites.

### User Story
As Super Admin, I want to be able to view/delete admin/volunteer invites to the platform.

### Solution (Core Elements)
Create pending invites modal to be accessed by clicking the "Invitations" button on the User Management page.

Contains modal title ("Pending Invitations") on top left, "X" button to exit modal on top right, and pending invitations table in the modal body (#480)

### Mockups

Other designs: https://www.figma.com/file/TWJTNWjK75A6VgSisySTLe/Sistering-%2F-Admin-Experience?node-id=1643%3A48816

### Technical
- For easier dev process, please complete #480 and #481 in this ticket as well
- You want a new `UserInvitesModal`  which will accept a list of `UserInviteDTO` which will contain `email` `createdAt` `role` and `uuid` in each `UserInviteDTO` (create this type if it does not exist)
- There are two table view which should be tracked by filtering whether the role in the DTO response is ADMIN or VOLUNTEER
- Please follow https://chakra-ui.com/docs/components/modal and https://chakra-ui.com/docs/components/table and https://chakra-ui.com/docs/components/tabs/usage for reference

### PR DESCRIPTION
**Pull request by briantu**
## Ticket link

Closes #479

## Implementation description
* Created `UserInvitesModal.tsx` and `UserInvitesTable.tsx` in `frontend/src/components/admin`
* Hooked up modal to "Invitations" button in `AdminUserManagementPageHeader.tsx`
* Created `UserInviteTypes.ts` and added `UserInviteDTO`
* Made some changes to the Table brand theme

## Steps to test
1. Go to `http://localhost:3000/admin/users`
2. Click on "Invitations" button to open the modal
3. Check modal looks good

## What should reviewers focus on?
* Code correctness
* Design correctness

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/admin/UserInvitesModal.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        <ModalOverlay />
+        <ModalContent maxW="1000px" h="700px" py="50px" px="70px">
+          <ModalHeader>
+            <Text textStyle="display-small-regular">Pending Invitations</Text>
+          </ModalHeader>
+          <ModalCloseButton />
+          <ModalBody overflow="auto">
+            <UserInvitesTable
+              invites={[
+                {
```
Comment (MatoPlus) [inline frontend/src/components/admin/UserInvitesModal.tsx:36]: Can you add a TODO comment to populate this?

**Review by MatoPlus — CHANGES_REQUESTED** [review]
Can you change the tabs such that it filters by role?

File: frontend/src/components/admin/UserInvitesTable.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                  _active={{
+                    bg: "transparent",
+                  }}
+                  icon={<CloseIcon color="text.default" boxSize="13px" />}
+                />
+              </Td>
+            </Tr>
+          ))}
+        </Tbody>
+      </Table>
```
Comment (MatoPlus) [inline frontend/src/components/admin/UserInvitesTable.tsx:94]: I think this should be insider the Tab component within a TabPanels?

**Review by MatoPlus — COMMENTED** [review]
Aside from request, this looks good to merge

File: frontend/src/components/admin/UserInvitesTable.tsx (hunk 2 above)
Comment (briantu) [reply frontend/src/components/admin/UserInvitesTable.tsx]: Done! I decided to use a state for the selected tab cuz I think it's easier than having two TabPanels with a Table in each.

File: frontend/src/components/admin/UserInvitesTable.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                    aria-label="Cancel invite"
+                    variant="ghost"
+                    _hover={{
+                      bg: "transparent",
+                    }}
+                    _active={{
+                      bg: "transparent",
+                    }}
+                    icon={<CloseIcon color="text.default" boxSize="13px" />}
+                  />
```
Comment (MatoPlus) [inline frontend/src/components/admin/UserInvitesTable.tsx:95]: FollowUP: we need to call delete user invite by email

**Review by MatoPlus — APPROVED** [review]
LGTM

**Pull request merged**